sklearnex/cluster/k_means.py

```diff
@@ -109,14 +109,11 @@ def _onedal_fit_supported(self, method_name, X, y=None, sample_weight=None):
                 _is_csr(X) and daal_check_version((2024, "P", 700))
             ) or not issparse(X)
 
-            _acceptable_sample_weights = True
-            if sample_weight is not None or not isinstance(sample_weight, numbers.Number):
-                sample_weight = _check_sample_weight(
-                    sample_weight, X, dtype=X.dtype if hasattr(X, "dtype") else None
-                )
-                _acceptable_sample_weights = np.allclose(
-                    sample_weight, np.ones_like(sample_weight)
-                )
```

Comment:
Instead of removing it I prefer to branch the logic for different versions of sklearn.
For the skl 1.5 we can skip this. Did you tried this?

Reply:
Interesting finding, branching doesn't help. The interpreter still spends time even if the branch is not taken. The behavior is same in daal4py.

Reply:
Updated with version check branching. Note that this is slightly different from what daal4py has. Anyway should work fine.

Reply:
> Interesting finding, branching doesn't help. The interpreter still spends time even if the branch is not taken. The proposed change aligns with daal4py.

i find that really unusual/ not sure how thats even possible. have you done any trace or profiling to show this?

Reply:
My assumption was wrong, the "or" was the problem.

Reply:
ahhhhh i see now. thanks!

Reply:
> check branching. Note that

Functionally, it should be aligned first with the stock sklearn. This is more prior. Thank you!